pennylane/transforms/defer_measurements.py

```diff
@@ -0,0 +1,166 @@
+            self.base_wire = wire_or_aux.base_wire
+            self.count = wire_or_aux.count + count + 1
+        else:
+            self.base_wire = wire_or_aux
+            self.count = count
+
+    def __hash__(self):
+        return hash((Aux, self.base_wire, self.count))
+
+    def __eq__(self, other):
+        if isinstance(other, Aux):
+            return self.base_wire == other.base_wire and self.count == other.count
+        return False
+
+    def __str__(self):
+        return f"Aux({repr(self.base_wire)},{repr(self.count)})"
+
+    def __repr__(self):
+        return f"Aux({repr(self.base_wire)},{repr(self.count)})"
+
+
+class WireRemapper:
+
+    def __init__(self):
+        self._altered_wires = set()
+        self._measured_wires = {}
+
+    def mark_altered(self, wire):
+        self._altered_wires.add(wire)
+
+    def get_mapped(self, base_wire):
+        assert not isinstance(base_wire, Aux)
+        if base_wire in self._measured_wires.keys():
+            return Aux(base_wire, self._measured_wires[base_wire])
+        return base_wire
+
+    def mark_altered_and_get_mapped(self, base_wire):
+        assert not isinstance(base_wire, Aux)
+        mapped = self.get_mapped(base_wire)
+        self.mark_altered(mapped)
+        return mapped
+
+    def mark_measured_and_get_mapped(self, base_wire):
+        assert not isinstance(base_wire, Aux)
+        if base_wire not in self._measured_wires.keys():
+            if base_wire in self._altered_wires:
+                self._measured_wires[base_wire] = 0
+                return Aux(base_wire)
+            else:
+                # self._altered_wires.add(base_wire)  # not sure if this should be here?
+                return base_wire
+        else:
+            if Aux(base_wire, self._measured_wires[base_wire]) in self._altered_wires:
+                self._measured_wires[base_wire] += 1
+                return Aux(base_wire, self._measured_wires[base_wire])
+
+
+
+def make_mid_circuit_measurements_terminal(tape):
+
+    ops = []
+    for op in tape.queue:
+        ops.append(op)
+
+    wr = WireRemapper()
+    new_ops_reversed = []
+    for op in reversed(ops):
+        # need to work in the reverse direction for this tape transform
+        if isinstance(op, qml.ops.mid_circuit_measure._MidCircuitMeasure):
+            wire = op.wires[0]
+            mapped_wire = wr.mark_measured_and_get_mapped(wire)
+            op._wires = Wires([mapped_wire])
+        else:
+            new_wires = []
+            for wire in op.wires:
+                new_wires.append(wr.mark_altered_and_get_mapped(wire))
+            op._wires = Wires(new_wires)
+
+        new_ops_reversed.append(op)
+
+    # reverse the tape back to original form
+    with QuantumTape() as new_tape:
+        for op in reversed(new_ops_reversed):
+            apply(op)
+
+    return new_tape
+
+
+def defer_measurements_on_mid_circuit_measured_terminal_tape(tape):
+
+    with QuantumTape() as new_tape:
```

Comment:
@puzzleshark just realized that since we're using `qfunc_transform`, tape creation is done implicitly, so we don't need to create a new one. Otherwise, we'd have a new nested tape and an extra dimension in the result.

See e.g., this [qfunc transform](https://pennylane.readthedocs.io/en/stable/_modules/pennylane/transforms/unitary_to_rot.html).